tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/paging/PagingSendTest.java

```diff
@@ -288,6 +296,408 @@ public void testPagingDoesNotDuplicateBatchMessagesAfterPagingStarted() throws E
+         assertTrue(queuePagingStore != null && queuePagingStore.isPageFull());
+
+         // the pages reach the limit
+         long existingPages = queuePagingStore.getNumberOfPages();
+         long maxPages = queuePagingStore.getPageLimitBytes() / queuePagingStore.getPageSizeBytes();
+
+         assertTrue(existingPages > maxPages);
+
+         // but messages still under limit
+         PageCursorProvider cursorProvider = queuePagingStore.getCursorProvider();
+         Long existingMessages = PageCursorProviderTestAccessor.getNumberOfMessagesOnSubscriptions(cursorProvider);
+         assertTrue(existingMessages < maxMessages, "existing " + existingMessages + " should be less than max " + maxMessages);
+
+         server.stop(true);
+         waitForServerToStop(server);
+
+         // restart the server the pageFull is still true
+         try {
+            server.start();
+            waitForServerToStart(server);
+
+            queue = server.locateQueue(queueAddr);
+
+            queuePagingStore = queue.getPagingStore();
+            assertTrue(queuePagingStore != null && queuePagingStore.isPageFull());
+
+            // but messages still under limit
+            cursorProvider = queuePagingStore.getCursorProvider();
+            existingMessages = PageCursorProviderTestAccessor.getNumberOfMessagesOnSubscriptions(cursorProvider);
+            assertTrue(existingMessages < maxMessages, "existing " + existingMessages + " should be still less than max " + maxMessages);
+
+            // now increase the pageLimitBytes to unblock procuder
+            addressSettings.setPageLimitBytes(Long.valueOf(size * 20));
+            server.getAddressSettingsRepository().addMatch(queueName, addressSettings);
+
+            // no longer page full
+            assertTrue(queuePagingStore != null && !queuePagingStore.isPageFull());
+
+            // now send more messages until pagefull
+            locator = createFactory(isNetty());
+            try (ClientSessionFactory csf = createSessionFactory(locator)) {
+               stop = false;
+               session = csf.createSession(true, true);
+               producer = session.createProducer(queueAddr);
+               while (!stop) {
+                  try {
+                     ClientMessage message = createMessage(session, messageSize, totalMessages, null);
+                     producer.send(message);
+                     totalMessages++;
+                  } catch (ActiveMQAddressFullException ex) {
+                     stop = true;
+                  }
+               }
+            }
+
+            // check the page full
+            assertTrue(queuePagingStore != null && queuePagingStore.isPageFull());
+
+            // current pages not exceeds the max pages
+            Long currentPages = queuePagingStore.getNumberOfPages();
+            maxPages = queuePagingStore.getPageLimitBytes() / queuePagingStore.getPageSizeBytes();
+            assertTrue(currentPages <= maxPages);
```

Comment:
Again can verify against pre-known value, and/or verify the getters are accurate.